parachains/runtimes/collectives/collectives-polkadot/src/impls.rs

```diff
@@ -105,29 +104,35 @@ where
 
 	fn vote_proposal(
 		who: AccountIdOf<T>,
-		proposal: HashOf<T>,
+		proposal_bounded: Bounded<ProposalOf<T, I>>,
 		index: ProposalIndex,
 		approve: bool,
 	) -> Result<bool, DispatchError> {
-		pallet_collective::Pallet::<T, I>::do_vote(who, proposal, index, approve)
+		pallet_collective::Pallet::<T, I>::do_vote(who, proposal_bounded, index, approve)
 	}
 
 	fn close_proposal(
-		proposal_hash: HashOf<T>,
+		proposal_bounded: Bounded<ProposalOf<T, I>>,
 		proposal_index: ProposalIndex,
 		proposal_weight_bound: Weight,
 		length_bound: u32,
 	) -> DispatchResultWithPostInfo {
 		pallet_collective::Pallet::<T, I>::do_close(
-			proposal_hash,
+			proposal_bounded,
 			proposal_index,
 			proposal_weight_bound,
 			length_bound,
 		)
 	}
 
-	fn proposal_of(proposal_hash: HashOf<T>) -> Option<ProposalOf<T, I>> {
-		pallet_collective::Pallet::<T, I>::proposal_of(proposal_hash)
+	fn proposal_of(proposal_bounded: Bounded<ProposalOf<T, I>>) -> Option<ProposalOf<T, I>> {
+		pallet_collective::Pallet::<T, I>::proposal_of(proposal_bounded)
+	}
+
+	fn bound_proposal(
+		proposal: ProposalOf<T, I>,
+	) -> Result<Bounded<ProposalOf<T, I>>, DispatchError> {
+		pallet_preimage::Pallet::<T>::bound(proposal)
```

Comment:
This is giving an error which can be solved by putting the bound `frame_system::Config<Hash = sp_core::H256>`, and it is pretty weird since Hash is set as `parachains_commons::types::Hash`, an alias of `sp_core::H256`.

<img width="1019" alt="image" src="https://github.com/paritytech/cumulus/assets/56828990/19669d2d-5ca9-4b6e-bf97-a4e68fab7b11">

Reply:
@ggwpez

Reply:
Why do you even need that custom `bound_proposal` function instead of injecting a `StorePreimage` trait into the pallet config trait?